PROTO_tests/tests/indexing_test.py

```diff
@@ -103,3 +103,8 @@ def test_bigint_indexing_preserves_max_bits(self):
         a = ak.arange(2**200 - 1, 2**200 + 11, max_bits=max_bits)
         assert max_bits == a[ak.arange(10)].max_bits
         assert max_bits == a[:].max_bits
+
+    def test_handling_bigint_max_bits(self):
+        a = ak.arange(2**200 - 1, 2**200 + 11, max_bits=3)
+        a[:] = ak.array(a, dtype=ak.bigint)
+        assert [7, 0, 1, 2, 3, 4, 5, 6, 7, 0, 1, 2] == a.to_list()
```

Comment:
So this does test the fix, but it's not as representative for bigint as I'd like.

This is because we are using the same array `a`, but this array has already been modded into the right range so instead of being `ak.arange(2**200 - 1, 2**200 + 11)` at time of indexing, it is already `[7, 0, 1, 2, 3, 4, 5, 6, 7, 0, 1, 2]`

On master it does in fact give a different result (which is good cause it means it tests the fix in this PR)
```python
In [3]: a = ak.arange(2**200 - 1, 2**200 + 11, max_bits=3)

In [4]: a
Out[4]: array([7 0 1 2 3 4 5 6 7 0 1 2])

In [5]: a[:] = ak.array(a, dtype=ak.bigint)

In [6]: a
Out[6]: array([1 0 1 2 0 1 2 0 1 0 1 2])
```

But I would rather it be `a[:] = ak.arange(2**200 - 1, 2**200 + 11)`

Reply:
Updated, good catch